pilot/pkg/model/push_context.go

```diff
@@ -311,20 +322,47 @@ func (ps *PushContext) UpdateMetrics() {
 
 // Services returns the list of services that are visible to a Proxy in a given config namespace
 func (ps *PushContext) Services(proxy *Proxy) []*Service {
-	// TODO: use network scopes here, and return services in the proxy namespace if
-	// configured to use current_namespace as the scope and proxy is not nil
-	//if proxy == nil {
-	//	return ps.allServices
-	//}
-	return ps.allServices
+	out := []*Service{}
+	for _, publicServices := range ps.publicServicesByNamespace {
+		out = append(out, publicServices...)
+	}
```

Comment:
this logic is incorrect. see comment below

Reply:
Oh, good catch

Reply:
done